File: Server/src/main/content/global/skill/production/cooking/WatermelonSlicePlugin.java
package content.global.skill.production.cooking;

import core.game.interaction.NodeUsageEvent;
import core.game.interaction.UseWithHandler;
import core.game.node.item.GroundItemManager;
import core.game.node.item.Item;
import core.plugin.Initializable;
import core.plugin.Plugin;

/**
 * The Watermelon slice plugin.
 */
@Initializable
public final class WatermelonSlicePlugin extends UseWithHandler {

    private static final Item KNIFE = new Item(946);

    private static final Item WATERMELON = new Item(5982);

    private static final Item WATERMELON_SLICE = new Item(5984);

    /**
     * Instantiates a new Watermelon slice plugin.
     */
    public WatermelonSlicePlugin() {
        super(KNIFE.getId());
    }

    @Override
    public Plugin<Object> newInstance(Object arg) throws Throwable {
        addHandler(5982, ITEM_TYPE, this);
        return this;
    }

    @Override
    public boolean handle(NodeUsageEvent event) {
        if (event.getPlayer().getInventory().remove(WATERMELON)) {
            for (int i = 0; i < 3; i++) {
                if (!event.getPlayer().getInventory().add(WATERMELON_SLICE)) {
                    GroundItemManager.create(WATERMELON_SLICE, event.getPlayer());
                }
            }
            event.getPlayer().getPacketDispatch().sendMessage("You slice the watermelon into three slices.");
        }
        return true;
    }

}

File: Server/src/main/content/global/skill/production/cooking/plugins/WatermelonSlicePlugin.kt
package content.global.skill.production.cooking.plugins

import core.game.interaction.NodeUsageEvent
import core.game.interaction.UseWithHandler
import core.game.node.item.GroundItemManager
import core.game.node.item.Item
import core.plugin.Initializable
import core.plugin.Plugin

@Initializable
class WatermelonSlicePlugin : UseWithHandler(KNIFE.id) {

    override fun newInstance(arg: Any): Plugin<Any> {
        addHandler(5982, ITEM_TYPE, this)
        return this
    }

    override fun handle(event: NodeUsageEvent): Boolean {
        if (event.player.inventory.remove(WATERMELON)) {
            for (i in 0..2) {
                if (!event.player.inventory.add(WATERMELON_SLICE)) {
                    GroundItemManager.create(WATERMELON_SLICE, event.player)
                }
            }
            event.player.packetDispatch.sendMessage("You slice the watermelon into three slices.")
        }
        return true
    }

    companion object {
        private val KNIFE = Item(946)
        private val WATERMELON = Item(5982)
        private val WATERMELON_SLICE = Item(5984)
    }
}
